third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -536,14 +536,14 @@ impl Interpreter {
     }
 
     /// Loads a resource from the data store and return the number of bytes read from the storage.
-    fn load_resource<'b>(
+    fn load_resource<'c>(
         loader: &Loader,
+        data_store: &'c mut TransactionDataCache,
```

Comment:
Wonder whether this is really an improvement, before the interpreter would work on any impl of a DataStore, now we specialize it to work on the cache directly.

Reply:
I do think conceptually this is an improvement, the value cache is important for performance and I think we would want that to be inlined/directly called. For me, intuitively, it seems clear that it's the actual lookup of data from the underlying (untyped) db that is the clear delineation for using dyn. Indeed the MoveResolver has different implementation for testing/prod etc. 

Previously the interpreter was using "impl DataStore" which would specialize for each implementation of TransactionDataCache<S> where S is the MoveResolver used. Because now we use dyn in TransactionDataCache, this has become a non-generic type and removing the DataStore trait allows TransactionDataCache to not carry the loader, which is already present as direct parameters.